src/main/java/com/klarna/hiverunner/data/InsertIntoTable.java

```diff
@@ -0,0 +1,191 @@
+package com.klarna.hiverunner.data;
+
+import java.io.File;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hive.hcatalog.api.HCatClient;
+import org.apache.hive.hcatalog.api.HCatTable;
+import org.apache.hive.hcatalog.common.HCatException;
+
+import com.klarna.hiverunner.HiveShell;
+
+/**
+ * A class for fluently creating a list of rows and inserting them into a table.
+ */
+public final class InsertIntoTable {
+
+  private final TableDataBuilder builder;
+  private final TableDataInserter inserter;
+
+  /**
+   * Factory method for creating an {@link InsertIntoTable}.
+   * <p/>
+   * This method is intended to be called via {@link HiveShell#insertInto(String, String)}.
+   *
+   * @param databaseName The database name.
+   * @param tableName The table name.
+   * @param conf The {@link HiveConf}.
+   * @return InsertIntoTable
+   */
+  public static InsertIntoTable newInstance(String databaseName, String tableName, HiveConf conf) {
+    TableDataBuilder builder = new TableDataBuilder(getHCatTable(databaseName, tableName, conf));
+    TableDataInserter inserter = new TableDataInserter(databaseName, tableName, conf);
+    return new InsertIntoTable(builder, inserter);
+  }
+
+  private static HCatTable getHCatTable(String databaseName, String tableName, HiveConf conf) {
+    HCatClient client = null;
+    try {
+      client = HCatClient.create(conf);
```

Comment:
Let's leave it as is, but we could consider moving the lifecycle of the HCatClient to a more central location in HiveRunner. All resources like this should preferably be handled at the "before" and "after" stages of the unit test. 
For one thing that makes it easier to assert that there is no leakage between test cases and that resources are cleaned up properly.